src/coreclr/src/jit/gentree.h

```diff
@@ -922,6 +928,8 @@ struct GenTree
 #define GTF_OVERFLOW                0x10000000 // Supported for: GT_ADD, GT_SUB, GT_MUL and GT_CAST.
                                                // Requires an overflow check. Use gtOverflow(Ex)() to check this flag.
 
+#define GTF_DIV_USE_MAGIC           0x80000000 // GT_DIV -- Uses MagicNumber multiplication to compute this division by a constant
```

Comment:
I am strongly against introducing any magic to our source base, `Use_Reciprocal_Mul_Or_Shift` or something similar to old `DIV_BY_CONST\DIV_USE_NO_DIV` looks better in the long term for me.

Reply:
Are commits about this magic division related to implementation of CSE and should be in this PR or could be extracted?

Reply:
I added this code to prevent  regressions.   It is a big win to perform a division by a constant using the reciprocal multiplication.  There were cases where we would CSE the constant and replace it with a CSE local variable.

I could extract the part of the change and check it in seperately.

Reply:
I factored out all of the prerequisite code changes for this into a new PR

#39021